packages/yswaps/scripts/libraries/solvers/multicall/Boo.ts

```diff
@@ -0,0 +1,107 @@
+  }
+
+  async solve({
+    strategy,
+    trades,
+    tradeFactory,
+  }: {
+    strategy: string;
+    trades: SimpleEnabledTrade[];
+    tradeFactory: TradeFactory;
+  }): Promise<PopulatedTransaction> {
+    const multicallSwapperAddress = (await ethers.getContract('MultiCallOptimizedSwapper')).address;
+    const strategySigner = await impersonate(this.strategyAddress);
+    const solidFromStrat = IERC20__factory.connect(this.solidAddress, strategySigner);
+
+    const multicallSwapperSigner = await impersonate(multicallSwapperAddress);
+    const solid = IERC20__factory.connect(this.solidAddress, multicallSwapperSigner);
+
+    const boo = IERC20__factory.connect(this.booAddress, multicallSwapperSigner);
+    const wftm = IWETH__factory.connect(this.wftmAddress, multicallSwapperSigner);
+    const spookyRouter = IUniswapV2Router02__factory.connect(this.spookyRouter, multicallSwapperSigner);
+
+    const solidBalance = (await solid.balanceOf(strategy)).sub(1);
+    console.log('[Boo] Total balance is', utils.formatEther(solidBalance), 'solid');
+
+    console.log('[Boo] Transfering solid to multicall swapper for simulations');
+    await solidFromStrat.transfer(multicallSwapperAddress, solidBalance);
+
+    // Create txs for multichain swapper
+    const transactions: PopulatedTransaction[] = [];
+
+    const path = [solid.address, wftm.address, boo.address];
+
+    console.log('[Boo] Getting sex => boo trade information');
+    const calculatedAmount = (await spookyRouter.getAmountsOut(solidBalance, path))[2];
+    const curveCalculatedTokenMinAmountOut = calculatedAmount.sub(calculatedAmount.mul(3).div(100)); // 3% slippage
+
+    console.log('[Boo] Expected boo out', utils.formatEther(calculatedAmount));
+
+    const approveSolid = (await solid.allowance(multicallSwapperAddress, spookyRouter.address)).lt(solidBalance);
+    if (approveSolid) {
+      console.log('[Boo] Approving solid');
+      const approveSolidTx = await solid.populateTransaction.approve(spookyRouter.address, constants.MaxUint256);
+      await multicallSwapperSigner.sendTransaction(approveSolidTx);
+      transactions.push(approveSolidTx);
+    }
+
+    console.log('[Boo] Executing solid => boo');
+    const sellSolidTx = await spookyRouter.populateTransaction.swapExactTokensForTokens(
+      solidBalance,
+      calculatedAmount,
+      path,
+      this.strategyAddress,
+      constants.MaxUint256
+    );
+    await multicallSwapperSigner.sendTransaction(sellSolidTx);
+    transactions.push(sellSolidTx);
+
+    const amountOut = await boo.balanceOf(this.strategyAddress);
+    console.log('[CurveYfiEth] Final boo balance is', utils.formatEther(amountOut));
+
+    if (amountOut.eq(0)) throw new Error('No boo tokens were received');
+
+    const data = mergeTransactions(transactions);
+
+    const executeTx = await tradeFactory.populateTransaction['execute((address,address,address,uint256,uint256)[],address,bytes)'](
```

Comment:
since theres only one tokenIn couldnt we use the "simple" execute for async trades that just receives ONE trade and not an array of them?

Reply:
Yes ! We should ! Nice catch.